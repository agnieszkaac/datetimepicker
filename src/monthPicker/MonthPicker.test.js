import React from "react";
import { render } from "@testing-library/react";

import { Month, MonthPicker } from "./";
import moment from "moment";

const onPick = jest.fn();
jest.mock("moment", () => (value) => ({
  month: (i) => "1",
  format: () => "Jan",
  isSame: (value, granularity) => true,
  toISOString: () => "",
}));

describe("<Month/> should", () => {
  it("math snapshot for NOT present month, NOT selected", () => {
    const { container } = render(
      <Month date={"2"} selected={false} onPick={onPick} />,
    );
    expect(container).toMatchSnapshot();
  });
  it("match snapshot for NOT present month, selected", () => {
    const { container } = render(
      <Month date={"2"} selected={true} onPick={onPick} />,
    );
    expect(container).toMatchSnapshot();
  });
  it("match snapshot for present month, NOT selected", () => {
    const { container } = render(
      <Month date={"1"} selected={false} onPick={onPick} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("match snapshot for present month, selected", () => {
    const { container } = render(
      <Month date={"1"} selected={true} onPick={onPick} />,
    );
    expect(container).toMatchSnapshot();
  });
});

describe("<MonthPicker /> should", () => {
  it("match snapshot", () => {
    const { container } = render(<MonthPicker onPick={onPick} />);
    expect(container).toMatchSnapshot();
  });
});
