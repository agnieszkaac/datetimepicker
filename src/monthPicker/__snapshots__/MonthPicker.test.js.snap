// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Month/> should match snapshot for NOT present month, selected 1`] = `
<div>
  <button
    class="month selected is-today"
    type="button"
    value=""
  >
    Jan
  </button>
</div>
`;

exports[`<Month/> should match snapshot for present month, NOT selected 1`] = `
<div>
  <button
    class="month is-today"
    type="button"
    value=""
  >
    Jan
  </button>
</div>
`;

exports[`<Month/> should match snapshot for present month, selected 1`] = `
<div>
  <button
    class="month selected is-today"
    type="button"
    value=""
  >
    Jan
  </button>
</div>
`;

exports[`<Month/> should math snapshot for NOT present month, NOT selected 1`] = `
<div>
  <button
    class="month is-today"
    type="button"
    value=""
  >
    Jan
  </button>
</div>
`;

exports[`<MonthPicker /> should match snapshot 1`] = `
<div>
  <div
    class="month-picker"
  >
    <button
      class="month is-today"
      type="button"
      value=""
    >
      Jan
    </button>
    <button
      class="month is-today"
      type="button"
      value=""
    >
      Jan
    </button>
    <button
      class="month is-today"
      type="button"
      value=""
    >
      Jan
    </button>
    <button
      class="month is-today"
      type="button"
      value=""
    >
      Jan
    </button>
    <button
      class="month is-today"
      type="button"
      value=""
    >
      Jan
    </button>
    <button
      class="month is-today"
      type="button"
      value=""
    >
      Jan
    </button>
    <button
      class="month is-today"
      type="button"
      value=""
    >
      Jan
    </button>
    <button
      class="month is-today"
      type="button"
      value=""
    >
      Jan
    </button>
    <button
      class="month is-today"
      type="button"
      value=""
    >
      Jan
    </button>
    <button
      class="month is-today"
      type="button"
      value=""
    >
      Jan
    </button>
    <button
      class="month is-today"
      type="button"
      value=""
    >
      Jan
    </button>
    <button
      class="month is-today"
      type="button"
      value=""
    >
      Jan
    </button>
  </div>
</div>
`;
